modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/translation_manager/TranslationAdminSelector.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayLayout from '@clayui/layout';
+import PropTypes from 'prop-types';
+import React, {useEffect, useMemo, useState} from 'react';
+
+import TranslationAdminModal from './TranslationAdminModal';
+
+const emptyArray = [];
+
+const TranslationAdminSelector = ({
+	activeLocales: initialActiveLocalesIds = emptyArray,
+	adminMode,
+	ariaLabels = {
+		default: Liferay.Language.get('default'),
+		manageTranslations: Liferay.Language.get('manage-translations'),
+		translated: Liferay.Language.get('translated'),
+		untranslated: Liferay.Language.get('untranslated'),
+	},
+	availableLocales = emptyArray,
+	defaultLocaleId,
+	selectedLocaleId: initialSelectedLocaleId,
+	showOnlyFlags,
+	small = false,
+	translations,
+}) => {
+	const [activeLocalesIds, setActiveLocalesIds] = useState(
+		initialActiveLocalesIds
+	);
+	const [selectedLocaleId, setSelectedLocaleId] = useState(
+		initialSelectedLocaleId
+	);
+	const [selectorDropdownActive, setSelectorDropdownActive] = useState(false);
+	const [translationModalVisible, setTranslationModalVisible] = useState(
+		false
+	);
+
+	const handleCloseTranslationModal = (activeLocalesIds) => {
+		setActiveLocalesIds(activeLocalesIds);
+		setTranslationModalVisible(false);
+	};
+
+	const activeLocales = useMemo(
+		() =>
+			availableLocales.filter((availableLocale) =>
+				activeLocalesIds.includes(availableLocale.id)
```

Comment:
Hey Diego,

soy you're asking to transform `activeLocalesIds` to a Set

```
const [activeLocalesIds, setActiveLocalesIds] = useState(
	new Set(initialActiveLocalesIds)
);
```

so we can use `has` instead of include, right?

If that's correct I don't think we're having a big optimization here, + we'll ned to change `TranslationAdminModal` API to fit with this new type.

Do you think this is worth?

Reply:
> a big optimization here

Yes, I also think about it. Since We have a very limited number of activeLocales.

> we'll ned to change TranslationAdminModal API to fit with this new type.

> Do you think this is worth?

I think We can leave it as an array 👍🏼

Reply:
So, the thing is:

- `availableLocales` prop is an array of objects.
- `activeLanguageIds` state is an array of strings (ids).
- `activeLocales` is an array of `availableLocales` which id is present in `activeLanguageIds`. We create this variable  to not have to iterate over `availableLocales` and check if is active on every render.

So, every time `availableLocales` or `activeLanguageIds` change we recalculate `activeLocales`.

To do what you suggest we need to:

- Create a Set `activeLanguageIdsSet` and update it every time `activeLanguageIds` changes.
```
const activeLanguageIdsSet = useMemo(() => {
	return new Set(activeLanguageIds);
}, [activeLanguageIds]);
```
- Replace `includes` with `has`. But we'll still have to iterate over `availableLocales` to find stuff, since it's an array of objects, remember.

What I guess yo were thinking is a map (dictionary) instead of a Set, so we can directly access to each locale via id, but that implies recreating that map every time `availableLocales` changes.

I'm not sure that adding all this complexity is worth since, as you pointed out, we're talking about 2300 operations in the worst case scenario (which takes aprox 149 ms)

Reply:
> But we'll still have to iterate over availableLocales to find stuff, since it's an array of objects, remember.

Yes 😢 

> I'm not sure that adding all this complexity is worth since, as you pointed out, we're talking about 2300 operations in the worst case scenario (which takes aprox 149 ms)

Yes, It will takes a time to do these changes for a better FPS. I think it's okay going with this solution and If someone relates a bug about it, We will already know a possible cause :p